src/schemes/fluid/fluid.jl

```diff
@@ -198,3 +198,23 @@ include("surface_tension.jl")
 include("surface_normal_sph.jl")
 include("weakly_compressible_sph/weakly_compressible_sph.jl")
 include("entropically_damped_sph/entropically_damped_sph.jl")
+
+@inline function update_speed_of_sound!(system, v, state_equation) end
+
+@inline function update_speed_of_sound!(system::WeaklyCompressibleSPHSystem, v,
+                                        state_equation::StateEquationAdaptiveCole)
+    max_velocity = 0
+
+    for particle in each_moving_particle(system)
+        tmp = dot(current_velocity(v, system, particle),
+                  current_velocity(v, system, particle))
+        if max_velocity < tmp
+            max_velocity = tmp
+        end
+    end
+
+    state_equation.sound_speed = min(state_equation.max_sound_speed,
+                                     max(state_equation.average_velocity,
```

Comment:
So `average_velocity` is just a lower bound for the sound speed? Then I think it makes much more sense to call it `min_sound_speed`, so that it is clear what it is doing.